pydantic/networks.py

```diff
@@ -464,22 +464,15 @@ def __init__(self, name: str, email: str):
     def __eq__(self, other: Any) -> bool:
         return isinstance(other, NameEmail) and (self.name, self.email) == (other.name, other.email)
 
-    @classmethod
```

Comment:
I am not sure if my approach is correct here.

Actually I don't know what this method is (can't find docs on the website). But seems deleting this method breaks the json schema generation on `format`. Currently unit test are failing and I need some guidance on fixing that.

I added support on `{"email": "", "name": ""}` schema for the input. Not sure if we should do that or just stick with plain `str` which is a bit easier to implement I guess.

Reply:
I think we can actually leave this here, that should help to resolve some of the issues with the failing tests.